### PR TITLE
Add new Raytac MDBT53-DB-40 and MDBT53V-DB-40 boards.

### PR DESCRIPTION
**Pull request by RaytacStanley**
Adds support for Raytac MDBT53-DB-40 and MDBT53V-DB-40 demo board.
These two boards pull out all the GPIOs of the module and connect them with pins without additional peripherals. It is very convenient for the development of various project programs.

### CODE REVIEW

File: boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+There are 2 samples that allow you to test that the buttons (switches) and LEDs on
+the board are working properly with Zephyr:
+
+   samples/basic/blinky
+   samples/basic/button
+
+You can build and flash the examples to make sure Zephyr is running correctly on
+your board. The button and LED definitions can be found in
+:zephyr_file:`boards/arm/raytac_mdbt53_db_40_nrf5340/raytac_mdbt53_db_40_nrf5340.dts`.
```
Comment (gmarull) [inline boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst:347]: this information is redundant, basic samples are well known and are expected to work with all boards that have buttons/leds.

File: boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+Pins can be configured in the board pinctrl file. To see the available mappings,
+open the `MDBT53-DB-40 Specification`_, chapter 2.5 'Pin Assignment'.
+Select the pins marked 'General-purpose I/O'. Note that pins marked as 'low-frequency I/O
+only' can only be used in under-10KHz applications. They are not suitable for SPI, I2C,
+UART, and PWM.
+
+References
+**********
+
+.. target-notes::
```
Comment (gmarull) [inline boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst:361]: is this needed? never seen that before

File: boards/arm/raytac_mdbt53v_db_40_nrf5340/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+the board are working properly with Zephyr:
+
+.. code-block:: console
+
+   samples/basic/blinky
+   samples/basic/button
+
+You can build and flash the examples to make sure Zephyr is running correctly on
+your board. The button and LED definitions can be found in
+:zephyr_file:`boards/arm/raytac_mdbt53v_db_40_nrf5340/raytac_mdbt53v_db_40_nrf5340.dts`.
```
Comment (gmarull) [inline boards/arm/raytac_mdbt53v_db_40_nrf5340/doc/index.rst:324]: ditto

File: boards/arm/raytac_mdbt53v_db_40_nrf5340/raytac_mdbt53v_db_40_nrf5340_cpunet_reset.c [hunk 4]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2023 Raytac Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
```
Comment (gmarull) [inline boards/arm/raytac_mdbt53v_db_40_nrf5340/raytac_mdbt53v_db_40_nrf5340_cpunet_reset.c:7]: nit: this file is not using any Kernel API

**Review by gmarull — COMMENTED** [review]
overall lgtm, a few nitpicks

File: boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst (hunk 1 above)
Comment (RaytacStanley) [reply boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst]: I have deleted this information.

File: boards/arm/raytac_mdbt53v_db_40_nrf5340/doc/index.rst (hunk 3 above)
Comment (RaytacStanley) [reply boards/arm/raytac_mdbt53v_db_40_nrf5340/doc/index.rst]: I have deleted this information.

**Comment by RaytacStanley** [comment]
Hi Gmarull; I modified the relevant question as you requested. The test has some errors, but I don’t know where the problem is. I want you to re-review the modified part first, so I proposed to re-review it again, but the others were removed. Can you please help me to see what the reason for the error in the twister test is and how to correct it?

**Comment by joerchan** [comment]
@RaytacStanley You have build errors for TF-M. Your board is not supported.
You should should remove the default enabling of TF-M on  your NS board and stop building the _ns board variant.

File: boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+
+.. note::
+
+   Depending on the TF-M configuration, an application DTS overlay may be
+   required, to adjust the Non-Secure image Flash and SRAM starting address
+   and sizes.
+
+2. Build the application firmware for the network core using
+   ``-DBOARD=raytac_mdbt53_db_40_nrf5340_cpunet``.
+
```
Comment (joerchan) [inline boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst:255]: .. note::
   Trusted Firmware-M (TF-M) and building the ``ns`` target is not supported for this board.

Comment (joerchan) [reply boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst]: Could you also add this note?
The other boards all have this note.

File: boards/arm/raytac_mdbt53v_db_40_nrf5340/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+Applications for the ``raytac_mdbt53v_db_40_nrf5340`` board configuration can be
+built, flashed, and debugged in the usual way. See :ref:`build_an_application` and
+:ref:`application_run` for more details on building and running.
+
+.. note::
+   Flashing and Debugging Zephyr onto the raytac_mdbt53v_db_40_nrf5340 board
+   requires an	external J-Link programmer. The programmer is attached to the J1
+   or J9 SWD connector.
+
+Building Secure/Non-Secure Zephyr applications with Arm |reg| TrustZone |reg|
```
Comment (joerchan) [inline boards/arm/raytac_mdbt53v_db_40_nrf5340/doc/index.rst:201]: Sorry, didn't see that it was 2 boards.
This should be similarly updated.

File: boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst (hunk 5 above)
Comment (RaytacStanley) [reply boards/arm/raytac_mdbt53_db_40_nrf5340/doc/index.rst]: I have added this note.

**Comment by RaytacStanley** [comment]
@joerchan  The twister test always fails on the samples/net/wpanusb. I reference the Thingy53 board to edit our MDBT53-DB-40 board and have the same problem. The thingy53 board builds samples/net/wpanusb in nRF connect SDK(NCS)  is ok. But, when I use zephyrprojcet's thingy53 board to build the samples/net/wpanusb in the NCS, it fails. I found thingy53 board is different in NCS and zephyrproject. The thingy53 board in NCS supports  TF-M, but the zephyrproject isn't. What do you think I should do?

**Comment by joerchan** [comment]
@RaytacStanley The build error occures because TF-M is disabled for the `_ns` board, and you can't enable it in zephyrproject because there is no valid TF-M board target for this. You should remove the `_ns` board variant in zephyrporject. It will only be supported in NCS.

**Comment by joerchan** [comment]
@RaytacStanley That does not make sense. You must have something strange in your setup. thingy53_nrf5340_cpuapp_ns does not compile. Are you running this is zephyr standalone or mixing in nRF Connect SDK?

Thingy53 fails, but because of a difference in BOARD_ENABLE_CPUNET default it fails differently, although with same reason.

Works:
`west build -p -b raytac_mdbt53_db_40_nrf5340_cpuapp samples/net/wpanusb`

Fails with missing soc_secure_mem_read.
`west build -p -b raytac_mdbt53_db_40_nrf5340_cpuapp samples/net/wpanusb`
`west build -p -b thingy53_nrf5340_cpuapp_ns samples/net/wpanusb -- -DCONFIG_BOARD_ENABLE_CPUNET=n`

Fails with Kconfig error on dependency for SOC_NRF_GPIO_FORWARDER_FOR_NRF5340
`west build -b thingy53_nrf5340_cpuapp_ns samples/net/wpanusb `

Looking into the sample.yaml I see the reason for why twister build would fail.
```
tests:
  sample.net.wpanusb:
    filter: dt_chosen_enabled("zephyr,ieee802154")
    platform_exclude: thingy53_nrf5340_cpuapp_ns
```
I think you need to add your board to the platform_exclude list, probably same for both samples.

**Comment by carlescufi** [comment]
@gmarull @anangl @joerchan please take another look

**Review by joerchan — DISMISSED** [review]
The TF-M configuration changes looks OK.
I have not reviewed the entire PR though, just TF-M related.

File: boards/arm/raytac_mdbt53v_db_40_nrf5340/raytac_mdbt53v_db_40_nrf5340_cpunet.dts [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+			label = "Green LED 0";
+		};
+		led1: led_1 {
+			gpios = <&gpio0 29 GPIO_ACTIVE_LOW>;
+			label = "Green LED 1";
+		};
+		led2: led_2 {
+			gpios = <&gpio0 30 GPIO_ACTIVE_LOW>;
+			label = "Green LED 2";
+		};
```
Comment (anangl) [inline boards/arm/raytac_mdbt53v_db_40_nrf5340/raytac_mdbt53v_db_40_nrf5340_cpunet.dts:42]: Documentation says that there is only one LED on this board. And its picture seems to confirm this.

**Review by anangl — CHANGES_REQUESTED** [review]

**Comment by RaytacStanley** [comment]
I am sorry to everyone; the update has been unable to upload due to a series of unknown issues. This has taken me two days. Therefore, I decided to re-establish a branch and launch a pull request to speed up.

**Pull request closed without merging**